app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -70,7 +73,54 @@ class ReturnsService @Inject() (
                                       returnToSubmit,
                                       returnId.appaId
                                     )
-      _                          <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))
+      _                          <- EitherT.right[ErrorResponse](
+                                      cacheRepository
+                                        .get(returnId)
+                                        .map {
+                                          case Some(ua) =>
+                                            auditReturnSubmitted(
```

Comment:
This arrangement will mean the cache is not cleared and the function doesn't return until the audit has taken place.

The for comprehension joins the stages together with flatMap and doing that means the futures have to complete in order.

In the case of audit, the requirement was fire and forget, so the process of creating the future as part of the audit will let it start in parallel and everything else should continue.

Thus this should be an _ = assignment rather than _ <- which also means you won't need to convert to an EitherT, and the clearing the repository should be the next line which does have to run in order so as the original line 73 - see comment below.